src/interpreter_funcs/extract.rs

```diff
@@ -0,0 +1,78 @@
+use std::error;
+use std::fmt;
+
+use crate::interpreter::{
+    Func, FuncError, FuncFlags, FuncInfo, ParamInfo, ParamRefinement, Ty, UintParamRefinement,
+    Value,
+};
+
+#[derive(Debug, PartialEq)]
+pub enum FuncExtractError {
+    Empty,
+}
+
+impl fmt::Display for FuncExtractError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Empty => write!(f, "No mesh geometry contained in group"),
+        }
+    }
+}
+
+impl error::Error for FuncExtractError {}
+
+pub struct FuncExtract;
+
+impl Func for FuncExtract {
+    fn info(&self) -> &FuncInfo {
+        &FuncInfo {
+            name: "Extract",
+            return_value_name: "Extracted Mesh",
+        }
+    }
+
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[
+            ParamInfo {
+                name: "Group",
+                refinement: ParamRefinement::GeometryArray,
+                optional: false,
+            },
+            ParamInfo {
+                name: "Index",
+                refinement: ParamRefinement::Uint(UintParamRefinement {
+                    default_value: Some(0),
+                    min_value: Some(0),
+                    max_value: None,
+                }),
+                optional: false,
+            },
+        ]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Geometry
+    }
+
+    fn call(&mut self, values: &[Value]) -> Result<Value, FuncError> {
+        let geometry_array = values[0].unwrap_geometry_array();
+        let index = values[1].unwrap_uint();
+
+        if geometry_array.is_empty() {
+            return Err(FuncError::new(FuncExtractError::Empty));
+        }
+
+        // FIXME: @Correctness The wrapping index is just a temporary
+        // crutch until we can report errors to the user.
+        let wrapping_index = index % geometry_array.len();
```

Comment:
I'm worried the user will get errors way too often due to such strictness.
We should at least consider treating all the cases when the min-max domain is dynamic this way (wrapping around) as a principle. We can then report a warning instead of an error. 
More generally, we should try to autofix each case when a parameter depends on the actual pipeline calculation and report a verbose warning suggesting a meaningful manual fix (i.e. suggesting a min-max range).

Reply:
The programmer in me agrees with your second suggestion. We should always give the user as much information as possible so that they can make good decisions and fix problems. I am just not sure what non-programmer people will expect :joy:

Reply:
The programmer and the user meet half way.
In this case the autofix will be the wrapping or clamping + a warning tells the user what to do to remove it.